### PR TITLE
[ci] rebalanced Windows tests

### PR DESCRIPTION
**Pull request by StrikerRUS**
Now we have the following setting for Windows:
 - VSTS: `regular` (with examples), `sdist`, `bdist` under **VS 2017** (~10 min)
 - Appveyor: `regular` (with examples) under **VS 2015** and **MinGW** (~10-15 min, was ~40 min)

Also, MinGW version has been updated from 6.3.0 to 7.3.0.

Known bug: conda at Windows cannot handle Python 3.7 properly yet (`Importing the multiarray numpy extension module failed.  Most likely you are trying to import a failed build of numpy. Original error was: DLL load failed: The specified module could not be found.` at VSTS and `Command executed with exception:   from collections import Sequence` at Appveyor).

While working on this PR, I found weird bug. The same symptoms were described in #1475. When I tried to run **all** examples by `pytest %APPVEYOR_BUILD_FOLDER%\tests` instead of `pytest %APPVEYOR_BUILD_FOLDER%\tests\python_package_test`, two C_API tests were passed succesfully, but all other tests failed with `OSError: exception: access violation writing 0xFFFFFFFF93400000`. At present this bug is avoided by running only Python tests (without C_API). I suppose, it's somehow connected to that the same `lib_lightgbm.dll` is loaded twice from different places (from installation directory for C_API tests and from python-package directory for Python tests).

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -54,13 +50,7 @@ test_script:
         if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
       }  # run all examples
   - IF "%COMPILER%"=="MINGW" appveyor exit  # skip all further steps
-  - cd %APPVEYOR_BUILD_FOLDER%\python-package && python setup.py bdist_wheel --plat-name=win-amd64 --universal
 
 artifacts:
-- path: Release/lib_lightgbm.dll
+- path: python-package\compile\windows\x64\DLL\lib_lightgbm.dll
   name: Library
```
Comment (StrikerRUS) [inline .appveyor.yml]: May I leave this artifact for my personal purpose? Due to some reason I have installed only VS 2013 on my machine, but sometimes need to test the latest library file from `master`.

Comment (guolinke) [reply .appveyor.yml]: sure

Comment (StrikerRUS) [reply .appveyor.yml]: Thanks!

**Comment by guolinke** [comment]
it seems azure pipeline is release and free for open source projects. maybe we can move on it?

**Comment by StrikerRUS** [comment]
Wow, how many CIs ! 😃 

My opinion is that we need to balance between our current CIs (VSTS, Appveyor, Travis), because we have enough experience in them, and after that give a try to Azure.

BTW, which OS are supported at Axure?

**Comment by guolinke** [comment]
@StrikerRUS I think Azure Pipeline is actually the vsts-ci, but it is free for open-source project.

**Comment by StrikerRUS** [comment]
Sorry, am I right, that you want to drop VSTS  and move to Azure?

**Comment by guolinke** [comment]
@StrikerRUS Yeah, I think they share the same config. But Azure Pipeline may is able to add community users to manage, while vsts-ci can't.

**Comment by StrikerRUS** [comment]
@guolinke Got it!

Seems Azure Pipelines is very young! 😃 They have been announced yesterday: http://disq.us/p/1vm7rh5

>I think they share the same config. 

Does it mean that I can continue rebalancing (at present I'm working on Linux, practically finished)? And then we'll need just to perform some minor management work, like register an account, rename the config file, etc?

**Comment by StrikerRUS** [comment]
Yesterday I noticed that the badge has been changed, now the name of service.

image

**Comment by guolinke** [comment]
@StrikerRUS yeah, the free version still has 10 concurrent jobs.

**Comment by guolinke** [comment]
@StrikerRUS just move to it, seems everything is fine.

**Comment by StrikerRUS** [comment]
@guolinke I've noticed it: 404 error happened just right when I wanted to check my rebalancing for Linux  status 😄 

GPU failed. Seems it doesn't perform `--recursive` clone.

**Comment by guolinke** [comment]
@StrikerRUS now you can admin the LightGBM-CI on Azure Pipeline 😄

**Comment by StrikerRUS** [comment]
@guolinke Thank you very much! Azure Pipeline seems to be more convenient than VSTS: there are no useless things on the screen, only important tabs and settings, and finally cancel and rebuild buttons!

**Comment by StrikerRUS** [comment]
@guolinke Hmm... I can rebuild PR from you, but cannot rebuild master.

image


image

**Comment by StrikerRUS** [comment]
@guolinke Also, it seems to be they use outdated gnutls library instead of openssl. Therefor, phantom error while pulling submodule happens.

```
fatal: unable to access 'https://github.com/Microsoft/LightGBM/': gnutls_handshake() failed: Error in the pull function.
```

**Comment by DavidStaheli** [comment]
Hi @StrikerRUS and @guolinke.  I'm a PM from the Azure Pipelines team.  Thanks a lot for giving it a try.  Please let me know if anything is missing that you need.  We want to help however we can.

**Comment by DavidStaheli** [comment]
We'll look into these and get each of them addressed.  Thanks a lot, @StrikerRUS.

**Comment by StrikerRUS** [comment]
@DavidStaheli Many thanks!

I've recollected another inconvenient case which consists in the fact that you need to run `chmod` before executing shell script (at both Ubuntu and macOS).
https://github.com/Microsoft/LightGBM/blob/6214657aed2c9c682e1937748d39d5bf78698164/.vsts-ci.yml#L94-L100

**Comment by DavidStaheli** [comment]
Thanks @StrikerRUS!  Looking into the chmod issue, I've seen reports of people needing to add the execute permission before committing & pushing the file, and then the git fetch will automatically re-apply the execute permission.  [Here's an example](https://stackoverflow.com/questions/40978921/how-to-add-chmod-permissions-to-file-in-git/40979016).  Do you know if that's already been done for your .sh files?  If so, it sounds like something _we_ need to address.

**Comment by StrikerRUS** [comment]
@DavidStaheli Thank you very much! The problem was solved using the solution from your link. It's rather strange that, for example, Travis doesn't suffer from the same problem. Anyway, thank you for the solution and sorry for disturbing you.

**Comment by brcrista** [comment]
@StrikerRUS I see you are running with V0 of the task.  V1 of the task doesn't require you to create an environment to use Conda.  So as a workaround, try upgrading to V1 of the task and make sure "Create a custom environment" isn't checked.

**Comment by StrikerRUS** [comment]
Thanks @brcrista for the reply! But could you please elaborate your workaround? Simple change `CondaEnvironment@0` to `CondaEnvironment@1` here
https://github.com/Microsoft/LightGBM/blob/64002b2c0de9d2308a5bb90e8e91fca184fcb744/.vsts-ci.yml#L82-L87
broke entire conda setup.

**Comment by brcrista** [comment]
Look at the YAML snippet at https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/package/conda-environment?view=vsts

I'm saying `createCustomEnvironment: false` and omit `environmentName`.  Let me know if that works

**Comment by StrikerRUS** [comment]
@brcrista I've already seen that snippet. Unfortunately, we need named environment in our test scripts, so I'd better leave my current hotfix till you'll release a real fix for the permission error.

**Comment by brcrista** [comment]
Can you try running on Hosted Linux Preview instead of Hosted Ubuntu 16.04?  I believe that should work.

**Comment by StrikerRUS** [comment]
@brcrista I'm facing this issue on macOS, not Linux:
https://github.com/Microsoft/LightGBM/blob/64002b2c0de9d2308a5bb90e8e91fca184fcb744/.vsts-ci.yml#L62-L87

**Comment by brcrista** [comment]
sorry, never mind then.

**Pull request merged**